### PR TITLE
Makes dwarf text not check every replacement every time.

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

Rather than do a for loop through 1500 replacements and running replaceTextEx on the line of speech every single time, making this one proc one of the top 10 laggiest in the entire codebase, it runs through each individual word in the line of speech and grabs the line from the file that way, making it run approximately 100 times faster.

## Why It's Good For The Game

Just because they're both O(n) doesn't mean one isn't faster.

## Changelog
:cl:
refactor: Dwarf speech is no longer absolutely paranoid about word replacement.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 1]
```diff
@@ -51,17 +51,16 @@ GLOBAL_LIST_INIT(dwarf_last, world.file2list("strings/names/dwarf_last.txt")) //
...
-				if(islist(value)) //Whether they like it or not.
-					value = pick(value) //This could be drastically reduced if needed though.
-				message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
-				message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
-				message = replacetextEx(message, " [key]", " [value]") //Also its scottish.
+	if(speech_args[SPEECH_LANGUAGE] != /datum/language/dwarf && message[1] != "*") // No accent if they speak their language
+		message = " [message]" //Credits to goonstation for the strings list.
+		var/list/dwarf_words = strings("dwarf_replacement.json", "dwarf") //thanks to regex too.
+		for(var/word in splittext(message," "))
+			var/value = dwarf_words[word] //Thus they will always be in character.
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species_types/dwarves.dm:58]:
```suggestion
			var/value = dwarf_words[word] //Thus they will always be in character.
			if(!value)
				continue
```

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 2]
```diff
@@ -51,17 +51,16 @@ GLOBAL_LIST_INIT(dwarf_last, world.file2list("strings/names/dwarf_last.txt")) //
...
+			if(islist(value)) //Whether they like it or not.
+				value = pick(value) //This could be drastically reduced if needed though.
+			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species_types/dwarves.dm]:
```suggestion
			message = replacetextEx(message, " [uppertext(word)]", " [uppertext(value)]")
```

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 3]
```diff
@@ -51,17 +51,16 @@ GLOBAL_LIST_INIT(dwarf_last, world.file2list("strings/names/dwarf_last.txt")) //
...
+			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species_types/dwarves.dm]:
```suggestion
			message = replacetextEx(message, " [capitalize(word)]", " [capitalize(value)]")
```

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 4]
```diff
@@ -51,17 +51,16 @@ GLOBAL_LIST_INIT(dwarf_last, world.file2list("strings/names/dwarf_last.txt")) //
...
+			message = replacetextEx(message, " [key]", " [value]") //Also its scottish.
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species_types/dwarves.dm]:
```suggestion
			message = replacetextEx(message, " [word]", " [value]") //Also its scottish.
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
Necessary changes to make it work. All good otherwise.

**Comment by silicons** [comment]
@Putnam3145 e

**Pull request merged**